core/shared/src/main/scala/zio/Scope.scala

```diff
@@ -121,6 +129,14 @@ object Scope {
           releaseMap.add(finalizer).unit
         def close(exit: => Exit[Any, Any]): UIO[Unit] =
           ZIO.suspendSucceed(releaseMap.releaseAll(exit, executionStrategy).unit)
+        def fork: UIO[Scope.Closeable] =
+          ZIO.uninterruptible {
+            for {
+              scope     <- Scope.make
+              finalizer <- releaseMap.add(scope.close(_))
```

Comment:
You could add something here: if `scope` is closed, then it is removed from `releaseMap`. Would lead to more efficient memory utilization.

Reply:
The `finalizer` returned by `releaseMap.add` runs the original finalizer if it has not already been run and removes it from the release map. So the `scope.addFinalizerExit(finalizer)` on the following line guarantees that when the inner scope is closed the finalized will be removed from the outer scope.